src/Provider/PointCloudProvider.js

```diff
@@ -193,6 +194,13 @@ export default {
                 bbox = new THREE.Box3(
                     new THREE.Vector3(cloud.boundingBox.lx, cloud.boundingBox.ly, cloud.boundingBox.lz),
                     new THREE.Vector3(cloud.boundingBox.ux, cloud.boundingBox.uy, cloud.boundingBox.uz));
+
+                // do we have normal information
+                const normal = Array.isArray(layer.metadata.pointAttributes) &&
+                    layer.metadata.pointAttributes.find(elem => elem.startsWith('NORMAL'));
+                if (normal) {
+                    layer.material.defines[normal] = 1;
+                }
```

Comment:
The more general question is: do we expect users to be able subclass `Material` and not `ShaderMaterial`?

Reply:
> that would prevent having a dependency on this specific material

IMHO, we don't need to impose ourselves  too much constraints. In the same way that TileMesh/LayeredMaterial is not replaceable by another mesh/material, we should totally rely on PointsMaterial to implement our features.

Reply:
I disagree, the less coupling, the more itowns is untangled, usable and extensible... I would rather set opt-in hint properties to materials so that materials could honor the hint if possible.

Reply:
On this instance, we don't really care. The only "coupling" was the fact layer.material.defines was undefined (!). The current code works with a basic three material.

Reply:
I completely agree here. (We'll talk again about it when we will want to have the same functionality with the same material applied to pointclouds parsed from PNTS, PLY, LAS, LAZ...)